homeassistant/components/shelly/select.py

```diff
@@ -37,14 +37,95 @@
 class RpcSelectDescription(RpcEntityDescription, SelectEntityDescription):
     """Class to describe a RPC select entity."""
 
+    method: str
+
+
+class RpcSelect(ShellyRpcAttributeEntity, SelectEntity):
+    """Represent a RPC select entity."""
+
+    entity_description: RpcSelectDescription
+    _id: int
+
+    def __init__(
+        self,
+        coordinator: ShellyRpcCoordinator,
+        key: str,
+        attribute: str,
+        description: RpcSelectDescription,
+    ) -> None:
+        """Initialize select."""
+        super().__init__(coordinator, key, attribute, description)
+
+        if self.option_map:
+            self._attr_options = list(self.option_map.values())
+
+        if hasattr(self, "_attr_name") and description.role != ROLE_GENERIC:
+            delattr(self, "_attr_name")
+
+    @property
+    def current_option(self) -> str | None:
+        """Return the selected entity option to represent the entity state."""
+        if not isinstance(self.attribute_value, str):
+            return None
+
+        if self.option_map:
+            return self.option_map[self.attribute_value]
+
+        return self.attribute_value
```

Comment:
We don't currently have an entity that uses this. This is the basis for future implementations.